webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/DataSourceController.scala

```diff
@@ -144,6 +145,7 @@ class DataSourceController @Inject()(
                                                       urlOrHeaderToken(token, request)) {
             for {
               (dataSourceId, datasetSizeBytes) <- uploadService.finishUpload(request.body)
+              _ <- dataSourceService.checkInbox(false)
```

Comment:
It should not be necessary here to do a full scan of the inbox. Instead, we should report the newly created dataset to the wk side individually. 

And in fact, I think this already happens in uploadService.finishUpload (it calls `dataSourceRepository.updateDataSource(dataSource)` which does exactly that)

Reply:
@philippotto You requested this change, did this do anything for you?

Reply:
The front-end code in `master` has a 3 seconds delay after a DS upload which I wanted to get rid of. After you did the change here, I removed the delay and uploading worked fine. However, it could be that the delay already was superfluous for some time. I temporarily removed the line locally and a test upload worked for me without problems.
So, if you think, that this line doesn't do anything and a finished upload-request implies that the DS is available (no race conditions exist), we can remove `checkInbox` again and keep the front-end-delay removed, too.
If you are not confident, I'd keep the 3s delay. I definitely want to avoid that a user uploads something, clicks on "view" (in a success-modal; not in the dashboard) and gets greeted with a 404.